src/content/blog/you-probably-don't-need-eslint-config-prettier-or-eslint-plugin-prettier/index.mdx

```diff
@@ -0,0 +1,216 @@
+> Running ESLint with [`--print-config`](https://eslint.org/docs/latest/use/command-line-interface#--print-config), can alternately print out the full list of lint rule configurations for a file.
+
+## `eslint-plugin-prettier`
+
+`eslint-plugin-prettier` is an ESLint _plugin_ that provides two things:
+
+- A custom rule, `prettier/prettier`, that runs all of Prettier inside a single ESLint rule
+- A shareable config, `plugin:prettier/recommended`, that enables the `prettier/prettier` rule
+
+For example, in ESLint's legacy config format, you might enable them like:
+
+```json
+{
+	// 1. Load in the prettier/prettier rule from the plugin
+	"plugins": ["prettier"],
+
+	// 2. Turn on the prettier/prettier rule
+	"extends": ["plugin:prettier/recommended"]
+}
+```
+
+The advantage of this approach is that you don't need to separately configure Prettier alongside ESLint.
+You can have one file -your ESLint config- that enables both.
+
+### Why `eslint-plugin-prettier` Is Often Harmful
+
+There are two big problems with running Prettier inside an ESLint rule the way `eslint-plugin-prettier` does:
+
+- Behavioral: it joins Prettier's reports with ESLint's, which in my experience confuses developers who aren't familiar with the tools
+- Performance: formatting is now blocked on all linting, which is often much slower than formatting
+
+The performance point can become quite bad in projects that utilize [type-checked rules](https://typescript-eslint.io/linting/typed-linting).
+
+- If `prettier/prettier` is the only lint rule that produces reports containing an auto-fixer, linting has to run twice
+- If any other rules introduce auto-fixes, one or more additional cycles may be introduced from `prettier/prettier` fixing formatting issues with those auto-fixes
+
+> 😬 Keep in mind that lint rules don't have visibility into formatting settings.
+> Their auto-fixers aren't likely to produce code that aligns to your formatter.
+
+Type-checked linting is by nature generally at least as slow as running the TypeScript type checker on all linted files.
+Running linting additional extra times adds up - and contributes to false negative perceptions around ESLint's and typescript-eslint's performance.
+
+**I strongly recommend you do not use `eslint-plugin-prettier`**.
+We even explicitly recommend against `eslint-plugin-prettier` in the [typescript-eslint formatting FAQs](https://typescript-eslint.io/linting/troubleshooting/formatting#eslint-plugin-prettier) and [typescript-eslint performance troubleshooting docs](https://typescript-eslint.io/linting/troubleshooting/performance-troubleshooting#eslint-plugin-prettier).
+
+If `prettier/prettier` is enabled in your ESLint configuration, the best step you can take is to remove it and uninstall the `eslint-plugin-prettier` package altogether.
+Failing that, you can disable the rule manually (`"prettier/prettier": "off"` under `"rules"`).
```

Comment:
It might be worth noting that extending from `plugin:prettier/recommended` will [also configure `eslint-config-prettier` too](https://github.com/prettier/eslint-plugin-prettier/blob/8210e44309b168c7c06185eae8249b2d0eb65815/eslint-plugin-prettier.js#L87).

So `{extends: ["plugin:prettier/recommended"], rules: {'prettier/prettier': 'off'}` is not the same thing as removing extending from plugin:prettier altogether.

Reply:
Good point, thanks!